tests/test_order_book.py

```diff
@@ -0,0 +1,246 @@
+MODIFICATION_TIME_PB = timestamp_pb2.Timestamp(seconds=1672574400)
+
+
+class MockElectricityTradingService(
+    electricity_trading_pb2_grpc.ElectricityTradingServiceServicer
+):
+    """A mock gRPC service to simulate historic vs real-time streams."""
+
+    @staticmethod
+    def _construct_public_order_book_record(
+        order_id: int,
+    ) -> electricity_trading_pb2.PublicOrderBookRecord:
+        return electricity_trading_pb2.PublicOrderBookRecord(
+            id=order_id,
+            delivery_area=delivery_area_pb2.DeliveryArea(
+                code="XYZ",
+                code_type=delivery_area_pb2.EnergyMarketCodeType.ENERGY_MARKET_CODE_TYPE_EUROPE_EIC,
+            ),
+            delivery_period=delivery_duration_pb2.DeliveryPeriod(
+                start=START_TIME_PB,
+                duration=delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_15,
+            ),
+            type=electricity_trading_pb2.OrderType.ORDER_TYPE_LIMIT,
+            side=electricity_trading_pb2.MarketSide.MARKET_SIDE_BUY,
+            price=price_pb2.Price(
+                amount=decimal_pb2.Decimal(value="100.00"),
+                currency=price_pb2.Price.Currency.CURRENCY_EUR,
+            ),
+            quantity=power_pb2.Power(mw=decimal_pb2.Decimal(value="5.00")),
+            execution_option=(
+                electricity_trading_pb2.OrderExecutionOption.ORDER_EXECUTION_OPTION_AON
+            ),
+            create_time=CREATE_TIME_PB,
+            update_time=MODIFICATION_TIME_PB,
+        )
+
+    def _mark_as_unimplemented(self, context: Any) -> None:
+        """Set the context to UNIMPLEMENTED."""
+        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
+        context.set_details("Method not implemented in the mock servicer.")
+
+    async def ReceivePublicOrderBookStream(  # pylint: disable=invalid-overridden-method
+        self,
+        request: electricity_trading_pb2.ReceivePublicOrderBookStreamRequest,
+        context: ServicerContext[
+            electricity_trading_pb2.ReceivePublicOrderBookStreamRequest,
+            electricity_trading_pb2.ReceivePublicOrderBookStreamResponse,
+        ],
+    ) -> AsyncIterator[electricity_trading_pb2.ReceivePublicOrderBookStreamResponse]:
+        """Send different data based on whether start_time is set."""
+        is_historic = request.HasField("start_time") and request.start_time.seconds > 0
+
+        if is_historic:
+            yield electricity_trading_pb2.ReceivePublicOrderBookStreamResponse(
+                public_order_book_records=[
+                    self._construct_public_order_book_record(1),
+                    self._construct_public_order_book_record(2),
+                ]
+            )
+            return
+
+        yield electricity_trading_pb2.ReceivePublicOrderBookStreamResponse(
+            public_order_book_records=[
+                self._construct_public_order_book_record(9),
+            ]
+        )
+        await asyncio.sleep(5)
```

Comment:
It might be a long time for a unit test. Can this sleep lower than 5 seconds?

Reply:
No, it's not too long because the asyncio.sleep(5) call is automatically cancelled and stopped as soon as the test finishes its assertions. Your test won't actually wait for the full 5 seconds.